lib/index.js

```diff
@@ -427,6 +487,37 @@ api.checkPermission = function(actor, permission, options, callback) {
     });
 };
 
+function _checkMembershipUpdate(actor, identity, groups, callback) {
+  // EXTERNAL: check to make sure IDENTITY_UPDATE_MEMBERSHIP permissions
+  //           exists for actor
+  // for each group in groups
+  //  check to make sure group exists
+  //  check to make sure group is a type: Group
+  //  x check to make sure owner is actor OR
+  //      actor has IDENTITY_UPDATE_MEMBERSHIP permission on resource??
+  async.each(groups, (group, callback) => {
+    async.auto({
+      get: callback => {
+        const query = {
+          id: database.hash(group),
+          'identity.type': 'Group',
+          'identity.sysStatus': 'active'
+        };
+        database.collections.identity.findOne(query, {
+          'identity.id': 1,
+          'identity.owner': 1
+        }, (err, record) => callback(err, record ? record.identity : null));
+      },
+      checkPermission: ['get', (callback, results) => {
+        // FIXME: is this permission check correct?
+        api.checkPermission(
+          actor, PERMISSIONS.IDENTITY_UPDATE_MEMBERSHIP,
+          {resource: results.get, translate: 'owner'}, callback);
```

Comment:
Yes, I believe this is correct. This is checking to see if the actor has at least one of these capabilities, where a capability is a permission + a resource:

1. update_membership + group_x
2. update_membership + owner_of_group_x

In effect, does the actor either:

1. Have the capability to update their membership with group x?
2. Have the capability to update their membership with any group owned by group x's owner?

The most common pattern will be to grant the second capability to most actors in the system -- where they are given the capability to update their membership with any group for which they are the owner.